intake/source/base.py

```diff
@@ -215,6 +222,27 @@ def to_spark(self):
         """
         raise NotImplementedError
 
+    @property
+    def entry(self):
+        if self._entry is None:
+            raise ValueError("Source was not made by a catalog")
+        if self._entry() is None:
+            raise RuntimeError("Tried to recover entry, but it was gone")
+        return self._entry()
+
+    def clone(self, **kwargs):
```

Comment:
What should the behaviour be if there is no entry, e.g., if this source was created with `intake.open_*`? Right now it will error, but I suppose it should make a new source anyway.

Reply:
In the worst case it could merge the kwargs it was created with with these kwargs (or just go with the defaults?) and return `type(self)(**{**my_kwargs, **kwargs})`?

I'm also not sure `clone` is the best name as one of the features of this is the ability to change it, however I don't have a better name.